cosmos_rl/rollout/vllm_rollout/monkey_patch_for_fp8.py

```diff
@@ -0,0 +1,79 @@
+import torch
+
+from torch.nn import Parameter
+
+from vllm.model_executor.layers.vocab_parallel_embedding import (
+    UnquantizedEmbeddingMethod,
+)
+from vllm.model_executor.layers.quantization.fp8 import Fp8KVCacheMethod
+from vllm.model_executor.layers.quantization.utils.w8a8_utils import Fp8LinearOp
+from vllm.model_executor.layers.quantization.fp8 import Fp8LinearMethod
+from vllm import _custom_ops as ops
+from vllm.model_executor.layers.quantization.utils import w8a8_utils
+
+
+"""
+This file is used to patch the vllm model to use rowwise fp8 linear.
+"""
+
+
+def apply_patch_to_dispatch():
+    # dispatch fp8 linear func to torch._scaled_mm per token/rowwise
+    def dispatch_fp8_linear_kernel_to_torch_scaled_mm(*args, **kwargs):
+        return w8a8_utils.torch_per_token_w8a8_scaled_mm
+
+    w8a8_utils.dispatch_w8a8_scaled_mm = dispatch_fp8_linear_kernel_to_torch_scaled_mm
+
+
+apply_patch_to_dispatch()
+
+
+def simplify_process_weights_after_loading():
+    def simplified_process_weights_after_loading(self, layer: torch.nn.Module) -> None:
+        # Warning: this is only for rowwise fp8 linear.
+        qweight, weight_scale = ops.scaled_fp8_quant(
+            layer.weight, scale=None, use_per_token_if_dynamic=True
+        )
+
+        # Update the layer with the new values
+        layer.weight = Parameter(qweight.t(), requires_grad=False)
+        layer.weight_scale = Parameter(weight_scale, requires_grad=False)
+        layer.input_scale = None
+
+    # modify the process_weights_after_loading method for rowwise fp8 linear.
+    Fp8LinearMethod.process_weights_after_loading = (
+        simplified_process_weights_after_loading
+    )
+
+
+simplify_process_weights_after_loading()
+
+
+# patch the Linear layer.
+def apply_fp8_linear_patch(model: torch.nn.Module):
+    for name, module in model.named_modules():
+        quant_method = getattr(module, "quant_method", None)
+        if quant_method is None:
```

Comment:
from line 56 to 77, do 
```
if isinstance(quant_method, Fp8LinearMethod):
   some_patching_here()
else:
    pass
```

do not import or refer any other types from vllm that we don't want to process right now